iron-doc-function.html

```diff
@@ -18,23 +18,55 @@
   <template>
     <style include="iron-doc-viewer-styles"></style>
     <style>
+      #funcName {
+        color: #212121;
+      }
+
       .paramName {
-        margin-right: 10px;
+        color: var(--iron-doc-accent-color1);
       }
 
-      #params marked-element,
-      #params .markdown-html,
-      #params .markdown-html p {
+      .paramType, .returnType {
+        color: var(--iron-doc-accent-color2);
+      }
+
+      #paramList .paramName {
+        color: var(--iron-doc-accent-color2);
+        margin-right: 5px;
+      }
+
+      #paramList marked-element,
+      #paramList .markdown-html,
+      #paramList .markdown-html p {
         display: inline;
       }
     </style>
 
-    <div id="signature">
-      <a href$="#[[anchorId]]" class="name deeplink">[[descriptor.name]]</a>(<span class="params code"><span>[[_paramText]]</span></span>)<!--
-      --><span class="return" hidden$="[[!descriptor.return]]">: <span class="type code">[[descriptor.return.type]]</span></span>
-    </div>
+    <code id="signature">
+      <a href$="#[[anchorId]]" id="funcName" class="name deeplink">
+        [[descriptor.name]]<!--
+   --></a><!--
+   -->(<span class="funcParams"><!--
+     --><template is="dom-repeat" items="[[descriptor.params]]" as="p"><!--
+       --><span hidden$="[[!index]]">, </span><!--
+       --><span class="paramName">[[p.name]]</span><!--
+       --><span hidden$="[[!p.type]]">:
+            <span class="paramType">[[p.type]]</span><!--
+       --></span><!--
+     --></template><!--
+   --></span>)<!--
+   --><span hidden$="[[!descriptor.return]]">:
```

Comment:
The effect I want here is e.g. `foo(bar, baz): qux` instead of `foo ( bar , baz ) : qux`. The issue is that if there is any whitespace between inline elements, a space will be rendered. `strip-whitespace` and `whitespace: normal` won't change that.

https://css-tricks.com/fighting-the-space-between-inline-block-elements/ shows a few approaches. The simplest is to put everything on one line, but I find that really hard to read and maintain. I like this commenting approach because it lets you format the HTML in the normal properly nested way, and then sprinkle in comments where you need to eliminate whitespace.